src/Hashids.net/Hashids.cs

```diff
@@ -207,6 +207,80 @@ public string EncodeLong(long number)
         /// <returns>Array of 64-bit integers.</returns>
         public long[] DecodeLong(string hash) => GetNumbersFrom(hash);
 
+        /// <summary>
+        /// Decodes the provided hash into numbers.
+        /// </summary>
+        /// <param name="hash">Hash string to decode.</param>
+        /// <returns>A 64-bit integer.</returns>
+        /// <exception cref="T:HashidsNet.MultipleResultsException">If the decoded hash returns more than one long</exception>
+        public long DecodeSingleLong(string hash)
+        {
+            var numbers = GetNumbersFrom(hash);
+
+            if (numbers.Length == 1) return numbers[0];
+            else throw new MultipleResultsException("The hash provided yielded more than one result.");
+        }
+
+        /// <summary>
+        /// Decodes the provided hash into numbers.
+        /// </summary>
+        /// <param name="hash">Hash string to decode.</param>
+        /// <param name="id">A 64-bit integer variable to output the result to.</param>
+        /// <returns>A 64-bit integer.</returns>
+        public bool DecodeSingleLong(string hash, out long id)
+        {
+            var numbers = GetNumbersFrom(hash);
+
+            if (numbers.Length == 1)
+            {
+                id = numbers[0];
+                return true;
+            }
+            else
+            {
+                id = -1;
+                return false;
+            }
+        }
+
+        /// <summary>
+        /// Decodes the provided hash into a number.
+        /// </summary>
+        /// <param name="hash">Hash string to decode.</param>
+        /// <returns>An integer.</returns>
+        /// <exception cref="T:System.OverflowException">If the decoded number overflows integer.</exception>
+        /// <exception cref="T:HashidsNet.MultipleResultsException">If the decoded hash returns more than one integer</exception>
+        public virtual int DecodeSingle(string hash)
+        {
+            var numbers = GetNumbersFrom(hash);
+
+            if (numbers.Length == 1) return (int)numbers[0];
+            else throw new MultipleResultsException("The hash provided yielded more than one result.");
+        }
+
+        /// <summary>
+        /// Decodes the provided hash into a number.
+        /// </summary>
+        /// <param name="hash">Hash string to decode.</param>
+        /// <param name="id">An integer variable to output the result to.</param>
+        /// <returns>An integer greater or equal to zero or -1 if the hash conversion yields more than one result.</returns>
+        /// <exception cref="T:System.OverflowException">If the decoded number overflows integer.</exception>
+        public virtual bool DecodeSingle(string hash, out int id)
+        {
+            var numbers = GetNumbersFrom(hash);
+
+            if (numbers.Length == 1)
+            {
+                id = (int)numbers[0];
+                return true;
+            }
+            else
+            {
+                id = -1;
```

Comment:
Returning the default value of int (`0`), would be more idiomatic and expected of a `TryXYZ(...)` API.
`int.TryParse(...)` for example returns `0`.

Also `-1` and `0` are _both_ valid return-values, so the caller needs to check the returned bool anyway to know if the out parameter has a valid value. Therefor sticking to `0` would be more the expected behavior.

```suggestion
                id = 0;
```

Reply:
Makes sense, to behave like int.TryParse, all set.